components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -162,6 +169,9 @@ struct IndexScanExecutorImpl {
     /// The strategy to decode handles.
     /// Handle will be always placed in the last column.
     decode_handle_strategy: DecodeHandleStrategy,
+
+    /// Whether need to decode partition id.
+    decode_partition_id: bool,
```

Comment:
Maybe you can store `pid_column_cnt` instead? `decode_partition_id` can be inferred from `pid_column_cnt > 0`.